Sources/TokamakDOM/Environment/DOMEnvironment.swift

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import JavaScriptKit
+import TokamakCore
+
+/// This is effectively a singleton, but a mounted `DOMEnvironment` is assumed to be a singleton
+/// too. It can't be declared as a property of `DOMEnvironment` as you can't modify it from within
+/// `body` callbacks.
+private var colorSchemeListener: JSClosure?
```

Comment:
Could you make it a `@State` then initialize it in the struct like so?


```swift
@State var colorSchemeListener = JSClosure {
  scheme = $0[0].object!.matches.boolean == true ? .dark : .light
  return .undefined
}
````

Reply:
I tried that, but every assignment on a `@State` property triggers a needless re-render, while this assignment to a private global doesn't.

Reply:
Yeah, but if you only assign it in the definition of the struct it won’t change on every render, right?

Reply:
oh nope nvm

Reply:
Ah, I see what you mean. That won't work because you're referencing `self` (implicitly via the reference to `scheme`) in the variable definition, `self` is not fully initialized yet at that point, which causes this error:

```
error: instance member 'scheme' cannot be used on type 'DOMEnvironment<V>'; 
did you mean to use a value of this type instead?
    scheme = $0[0].object!.matches.boolean == true ? .dark : .light
    ^~~~~~
```